### PR TITLE
article on element existence

### PR DESCRIPTION
**Pull request by titusfortner**
Please review for correct content in this. Should be able to push out Watir 6.12 once I can reference this article.

Thanks!

### CODE REVIEW

File: _posts/2018-07-17-element-existentialism.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+### What Matters
+
+There are a bunch of different terms that mean slightly different things when it comes to whether an element "exists".
+See the [Terminology section](#terminology) below for the meaning of the terms used in quotes in this article.
+
+There are two major uses cases for evaluating whether an element is. 
+1. Ensure it is usable before taking an action on it
+* This is what we implemented as the default for Watir 6. We intentionally maintained a distinction 
+between taking an action on an element and locating the element. The goal is to allow the instantaneous querying of 
+the the condition or status of an element at any time, while also making every effort to take the requested action. If
```
Comment (kapoorlakshya) [inline _posts/2018-07-17-element-existentialism.md]: > the the

Typo? :)

File: _posts/2018-07-17-element-existentialism.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+2. Validate the user can see it
+* For this we don't actually care if an element "exists" in the DOM, if it isn't also "displayed," so we need to explore
+what this means.
+
+Justin Ko has a [great synopsis](https://jkotests.wordpress.com/2012/11/02/checking-for-an-element-exists-visible-present/)
+of the three applicable predicate methods for a Watir element. Of these three, "exist" is typically insufficient
+for our needs and "visible" is redundant. In the US Legislature, a Representative has three options for a vote: 
+"aye", "no," and "present." I like that this connotation of "present" is the one that Watir chose to use, as it 
+indicagtes that the member is in plain sight on the floor of Congress and in a state ready and able to take action.
```
Comment (kapoorlakshya) [inline _posts/2018-07-17-element-existentialism.md]: > indicagtes 

Typo

File: _posts/2018-07-17-element-existentialism.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+##### Q: Does `#present?` or `#visible?` change if I scroll the element out of view?
+A: No. These methods are ultimately implemented by the atom `isShown`, and only attempts to answer the question of whether it 
+ can be seen by the user assuming that the element is moved into the viewport first, even if it has not been.
+
+##### Q: If `#present?` returns `true`, does that mean the element is interactable?
+A: Unfortunately, not necessarily. It is possible that attempting to click the element will give a "click intercepted" error 
+or an "element not interactable" error. Future versions of the w3c specification are likely going to provide
+ endpoints to evaluate whether this is true without taking the action, and future versions of Watir will make it easier
+ to set default handling behaviors for these issues. At this point it might still be messy, so 
+ []reach out to us](http://watir.com/help/) if you get stuck trying to handle this situation.
```
Comment (kapoorlakshya) [inline _posts/2018-07-17-element-existentialism.md]: > []

should be

> [

File: _posts/2018-07-17-element-existentialism.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+you rescue `UnknownObjectException` or `UnknownFrameException` in your code. While we aren't officially 
+deprecating `#visible?`, we recommend that you make sure that you need to be using any existence check in the first place 
+(see point #1 in the next section), and if you do, switch to using `#present?` instead.
+
+### What Matters
+
+There are a bunch of different terms that mean slightly different things when it comes to whether an element "exists".
+See the [Terminology section](#terminology) below for the meaning of the terms used in quotes in this article.
+
+There are two major uses cases for evaluating whether an element is.
```
Comment (p0deje) [inline _posts/2018-07-17-element-existentialism.md]: s/uses/use

Comment (jkotests) [reply _posts/2018-07-17-element-existentialism.md]: End should also be "exists" instead of "is".

**Comment by titusfortner** [comment]
merged already

**Pull request closed without merging**